com.unity.netcode.gameobjects/Runtime/NetworkVariable/Collections/NetworkSet.cs

```diff
@@ -1,18 +1,19 @@
 #if !NET35
-using System.Collections;
-using System.Collections.Generic;
+using System;
 using System.IO;
+using Unity.Collections;
 
 namespace Unity.Netcode
 {
     /// <summary>
     /// Event based NetworkVariable container for syncing Sets
     /// </summary>
     /// <typeparam name="T">The type for the set</typeparam>
-    public class NetworkSet<T> : NetworkVariableBase, ISet<T> where T : unmanaged
+    public class NetworkSet<T> : NetworkVariableBase where T : unmanaged, IEquatable<T>
```

Comment:
I'd second what Cosmin asked/commented in the PR. I'd appreciate a description explaining motivations behind dict/set and maybe in addition to that a quick summary of changes under this PR (such as dropping `ISet<T>` constraint from the interface etc.)

(oh, also, are we unit-testing all of the remaining methods now?)